test/GPv2Signing.test.ts

```diff
@@ -461,5 +496,59 @@ describe("GPv2Signing", () => {
       ).to.be.reverted;
       expect(await evilVerifier.state()).to.equal(ethers.constants.One);
     });
+
+    it("should verify pre-signed order", async () => {
```

Comment:
Maybe add a test case, that another order from the same user with just one small difference cannot be verified using the same pre-signed order.